From 7cd490fe7be65aedaec399648a5b47ac2e5e0780 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Thu, 3 Dec 2015 17:06:15 +0800
Subject: [PATCH 05/13] atomisp2: add atomisp_css_set_ob2_config()

OB2 is newly added for ISP2.7, add the new function to pass OB2
config to CSS.

Change-Id: Id8012883718840cf7404ef36d28209252ef905f9
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-13009
Original-Tracked-On: #H1604058414
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/87947
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M COHO (SC)
Reviewed-by: Cary J Gee <cary.j.gee@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/450386
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c        | 6 ++++++
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h        | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 6c21277..9e4de4e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -3880,6 +3880,12 @@ int atomisp_css_get_ob_config(struct atomisp_sub_device *asd,
 	return 0;
 }
 
+void atomisp_css_set_ob2_config(struct atomisp_sub_device *asd,
+			struct atomisp_css_ob2_config *ob2_config)
+{
+	asd->params.config.ob2_config = ob2_config;
+}
+
 int atomisp_css_get_dp_config(struct atomisp_sub_device *asd,
 			struct atomisp_dp_config *config)
 {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
index b62ad90..95588c8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.h
@@ -69,6 +69,7 @@
 #define atomisp_css_anr_config	ia_css_anr_config
 #define atomisp_css_xnr_config	ia_css_xnr_config
 #define atomisp_css_macc_config	ia_css_macc_config
+#define atomisp_css_ob2_config	ia_css_ob2_config
 #define atomisp_css_gamma_table	ia_css_gamma_table
 #define atomisp_css_ctc_table	ia_css_ctc_table
 #define atomisp_css_macc_table	ia_css_macc_table
@@ -260,6 +261,9 @@ void atomisp_css_unload_firmware(struct atomisp_device *isp);
 void atomisp_css_set_dvs_6axis(struct atomisp_sub_device *asd,
 			struct atomisp_css_dvs_6axis *dvs_6axis);
 
+void atomisp_css_set_ob2_config(struct atomisp_sub_device *asd,
+			struct atomisp_css_ob2_config *ob2_config);
+
 unsigned int atomisp_css_debug_get_dtrace_level(void);
 
 int atomisp_css_debug_dump_isp_binary(void);
-- 
1.9.1

